cvlab/view/widgets.py

```diff
@@ -407,6 +355,59 @@ def deleteLater(self):
             self.close_image_dialog()
 
 
+class OutputPreview(QHBoxLayout):
+    default_image = None
+    preview_callbacks = [
+        (np.ndarray, ActionImage.set_image),
+        (str, ActionImage.set_text),
+        (bool, ActionImage.set_bool)
+    ]
+
+    def __init__(self, output, previews_container):
+        super(OutputPreview, self).__init__()
+        if not self.default_image:
+            self.default_image = QPixmap(CVLAB_DIR + "/images/default.png")
+        self.output = output
+        self.previews_container = previews_container
+        self.setAlignment(QtCore.Qt.AlignCenter)
+        self.setContentsMargins(0, 0, 0, 0)
+        self.base_spacing = 4
+        self.previews = []
+        self.previews.append(ActionImage(self))
+        self.img = self.default_image
+        self.previews[0].setPixmap(self.img)
+        self.addWidget(self.previews[0])
+
+    def update(self, forced=False):
+        objects = self.get_preview_objects()
+        if not objects:
+            objects = [None]
+        self.adjust_number_of_previews(objects)
+        for i, obj in enumerate(objects):
+            if forced or self.previews_container.isVisible() or self.previews[i].image_dialog is not None:
+                for _type, callback in self.preview_callbacks:
```

Comment:
nie można było zrobić preview_callbacks jako słownika, żeby nie musieć iterować po liście i odwoływać się przez isinstance? chyba że to związane, z tym jak działają callbacki

Reply:
można, ale wtedy trzeba by dla każdego typu modelu dodawać wpis w słowniku, a tak to isinstance łapie wszystkie, po za tym na razie są tylko 4 typy więc te iteracje nie obciążą cvlaba